### ISSUE TITLE
NPY201 does not catch `ptp` updates for numpy v2

### ISSUE THREAD

**Issue by jvavrek** [astral-sh/ruff#11909]
`ruff` does not catch the `arr.ptp() --> np.ptp(arr)` updates necessary in `numpy>=2`

MWE:
```py
# tmp.py
import numpy as np

x = np.arange(10)
print(x.ptp())
```

```sh
ruff check tmp.py --select NPY201 --isolated  # (same without the --isolated flag)
# All checks passed!
```

```sh
ruff --version
# ruff 0.4.9
```

**Comment by dhruvmanila** [comment]
Thank you for highlighting this! I think this is difficult with the current semantic model as it requires Ruff to understand that the type of the variable `x` is a NumPy array. Related to https://github.com/astral-sh/ruff/issues/6432, if Ruff were to just check for the `.ptp` method, it would raise false positives if the variable is not a Numpy array.
